### PR TITLE
docs: bunch of fixes and improvements

### PR DESCRIPTION
**Pull request by mkniewallner**
## Summary

A bunch of fixes and improvements on different parts of the documentation. For Docker documentation changes, links to the relevant documentations have been included in the commit messages.

### CODE REVIEW

File: docs/guides/docker.md [hunk 1]
```diff
@@ -13,9 +13,8 @@ docker run ghcr.io/astral-sh/uv --help
 uv can be installed by copying from the official Docker image:
 
 ```dockerfile
-FROM ghcr.io/astral-sh/uv:latest as uv
 FROM python:3.12-slim-bullseye
-COPY --from=uv /uv /bin/uv
+COPY --from=ghcr.io/astral-sh/uv:latest /uv /bin/uv
```
Comment (mkniewallner) [inline docs/guides/docker.md:17]: This change might be controversial, as maybe we actually prefer to rely on a named stage?

File: docs/guides/using-azure-artifacts.md [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
 
 ## Authenticate using a PAT
-If you have a Personal Access Token available (eg [`$(System.AccessToken)` in an Azure pipeline](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#systemaccesstoken)), you can authenticate to Artifacts using Basic Auth. Simply include the PAT in the password field of the URL. The username can be any dummy string.
+
+If you have a Personal Access Token available (e.g. [`$(System.AccessToken)` in an Azure pipeline](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#systemaccesstoken)), you can authenticate to Artifacts using Basic Auth. Simply include the PAT in the password field of the URL. The username can be any dummy string.
 
 ```bash
 # if your token is in the ADO_PAT environment variable
-export UV_EXTRA_INDEX_URL=https://dummy:$ADO_PAT@pkgs.dev.azure.com/{organisation}/{project}/_packaging/{feedName}/pypi/simple/
+export UV_EXTRA_INDEX_URL=https://dummy:$ADO_PAT@pkgs.dev.azure.com/{organization}/{project}/_packaging/{feedName}/pypi/simple/
```
Comment (mkniewallner) [inline docs/guides/using-azure-artifacts.md:11]: "organisation" is valid in UK English, but I believe the documentation often uses US English, so switching to "organization" should improve consistency.

File: docs/guides/using-azure-artifacts.md [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
-
+uv can install packages from [Azure DevOps Artifacts](https://learn.microsoft.com/en-us/azure/devops/artifacts/start-using-azure-artifacts?view=azure-devops&tabs=nuget%2Cnugetserver). You can authenticate to a feed using a [Personal Access Token](https://learn.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate?view=azure-devops&tabs=Windows) or interactively using Keyring.
 
 ## Authenticate using a PAT
-If you have a Personal Access Token available (eg [`$(System.AccessToken)` in an Azure pipeline](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#systemaccesstoken)), you can authenticate to Artifacts using Basic Auth. Simply include the PAT in the password field of the URL. The username can be any dummy string.
+
+If you have a Personal Access Token available (e.g. [`$(System.AccessToken)` in an Azure pipeline](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#systemaccesstoken)), you can authenticate to Artifacts using Basic Auth. Simply include the PAT in the password field of the URL. The username can be any dummy string.
```
Comment (mkniewallner) [inline docs/guides/using-azure-artifacts.md:7]: Besides the newline, there is one typo fixed here: "e.g." instead of "eg".

**Review by charliermarsh — APPROVED** [review]
Thanks! These all look good to me.

**Pull request merged**